api/go.mod

```diff
@@ -0,0 +1,15 @@
+module github.com/nmstate/kubernetes-nmstate/api
+
+go 1.16
+
+require (
+	github.com/nmstate/kubernetes-nmstate/pkg/names v0.0.0
+	github.com/onsi/ginkgo v1.16.5
+	github.com/onsi/gomega v1.17.0
+	k8s.io/api v0.22.3
```

Comment:
This is not aligend with main module https://github.com/nmstate/kubernetes-nmstate/blob/main/go.mod#L27

Reply:
it was updated in both places automatically by `go mod tidy`, is there a reason to have `k8s.io/api v0.21.1` ?

Reply:
Right we are testing towards k8s 1.21.

Reply:
Ok, let's go with it, CI is fine and we are going to move to 1.22 soon.

Reply:
Looks like we have to "replace" to pin to the version:

```
+ GOARCH=amd64
+ go build -o /tmp/knmstate/kubernetes-nmstate/build/_output/bin//handler.manager.linux-amd64 ./cmd/handler
# k8s.io/client-go/applyconfigurations/apiserverinternal/v1alpha1
vendor/k8s.io/client-go/applyconfigurations/apiserverinternal/v1alpha1/storageversion.go:62:34: not enough arguments in call to managedfields.ExtractInto
	have (*v1alpha1.StorageVersion, typed.ParseableType, string, *StorageVersionApplyConfiguration)
	want (runtime.Object, typed.ParseableType, string, interface {}, string)
# k8s.io/client-go/applyconfigurations/policy/v1

```